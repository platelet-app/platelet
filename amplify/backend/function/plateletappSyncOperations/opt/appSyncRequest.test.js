const { request, errorCheck } = require("./appSyncRequest");
const { default: fetch, Request } = require("node-fetch");

jest.mock("node-fetch", () => ({
    ...jest.requireActual("node-fetch"),
    Request: class {
        constructor(input, init) {
            this.input =
                "https://xxxxxx.appsync-api.ap-northeast-1.amazonaws.com/graphql";
            this.init = "init";
        }
    },
    default: jest.fn().mockResolvedValue(
        Promise.resolve({
            json: jest.fn().mockResolvedValue({
                data: {
                    listTodos: {
                        items: [
                            {
                                id: "1",
                                name: "Todo 1",
                            },
                            {
                                id: "2",
                                name: "Todo 2",
                            },
                        ],
                    },
                },
            }),
        })
    ),
}));

describe("appSyncRequest", () => {
    test("make a request", async () => {
        const response = await request(
            {
                query: `query listTodos {
                listTodos {
                  items {
                    id
                    name
                  }
                }
              }`,
            },
            "https://xxxxxx.appsync-api.ap-northeast-1.amazonaws.com/graphql"
        );
        expect(response).toBeDefined();
        expect(fetch).toHaveBeenCalledTimes(1);
        expect(fetch).toMatchSnapshot();
    });
    test("error check function", async () => {
        const body = {
            errors: [
                {
                    message: "Error message",
                },
            ],
        };
        expect(() => errorCheck(body)).toThrow("Error message");
    });
});
