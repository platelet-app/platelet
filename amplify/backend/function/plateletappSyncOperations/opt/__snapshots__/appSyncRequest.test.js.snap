// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appSyncRequest make a request 1`] = `
[MockFunction] {
  "calls": [
    [
      Request {
        "init": "init",
        "input": "https://xxxxxx.appsync-api.ap-northeast-1.amazonaws.com/graphql",
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;
